MU-new page

hiroki-norota opened the pull request:
# What
商品出品ページのマークアップを実装する。
- routesにてnewページが表示されるよう編集。
- views/items配下にnewページおよびrenderを使用し、関連のhtml.hamlを作成。
- scssではapplication.scssを追記。
- フォルダー(pages・modules・mixin)にて各種scssを作成。

# Why
商品出品ページのマークアップはアプリケーションの必須機能であるため。
http://localhost:3000/items/new

springgarden29 commented:
optional trueを削除しました

hiroki-norota committed:
MU_newpage_norota

ghost commented:
Warnings are found on analyzing the commit 0c428a6.

**2** warnings:
- ⚠️ [Goodcheck](https://sider.review/gh/repos/265854956/pulls/13/logs/2355445c-623b-4fd4-baab-af604e35cfdc)
- ⚠️ [Querly](https://sider.review/gh/repos/265854956/pulls/13/logs/bf52da8c-9297-460f-9be5-5a31c7316e3a)

We recommend to address them as possible, for example, update outdated dependencies, fix the tool's configuration, [configure `sider.yml`](https://help.sider.review/getting-started/custom-configuration), [turn off unused tools](https://help.sider.review/getting-started/repository-settings), and so on.

If you are struggling with these errors or warnings, feel free to ask us via chat. 💬

shry44 commented on app/assets/stylesheets/pages/_newpage_items_new.scss:
```diff
@@ -0,0 +1,273 @@
+* {
+  box-sizing: border-box;
+}
```
全ページに影響が出るため、トップページが崩れるようです。
wrapperの中に記述すればこのページ内のみ影響するように修正できます。
他のページと区別のためwrapperのクラス名も念のため変えた方が良いかもしれません。

jikutkgtaku commented on app/views/items/_form_menu.html.haml:
```diff
@@ -0,0 +1,122 @@
+            %span.form-required 必須
+          = f.select :condition_id, [["選択してください",""],["レディース", 1], ["メンズ", 2], ["ベビーキッズ", 3],],{},{class: "select-box"}
+        .items-date--size.contents-hight
+          %h3
+            サイズ
+            %span.form-required 必須
+          = f.select :condition_id, [["選択してください",""],["S", 1], ["M", 2], ["L", 3],],{},{class: "select-box"}
```
これ見て気がついたんですけど、サイズの欄にSML以外のやつも入れた方が良いかと？(データベース関連は相談済み？）

jikutkgtaku commented on app/views/items/_form_menu.html.haml:
```diff
@@ -0,0 +1,122 @@
+          %h3
+            発送までの日数
+            %span.form-required 必須
+          = f.select :shipping_days, [["選択してください",""],["1~2日で発送", 1], ["2~3日で発送", 2],["4~7日で発送", 3]],{},{class: "select-box"}
```
active hashの使用はどうでしょうか？

jikutkgtaku reviewed (commented):
個人的に最低限のviewとしては問題ないかと思います。
コードについては一部思ったことを書き込みます。

springgarden29 commented:
私は皆さんが指摘している箇所以外はLGTMです。

hiroki-norota committed:
MU-new page fix 1

クラス名の変更　wrapper→form-new
form-newクラスにbox-sizing border-boxを設定
商品の詳細・配送についてを修正
商品名とブランドのtext-filedをcssを修正後、mixinでitem-name-boxを作成/変更
ロゴをクリック後トップページへ移動するようにリンクの設定

eclair6632 commented on app/assets/stylesheets/pages/_newpage_items_new.scss:
```diff
@@ -0,0 +1,245 @@
+$safecolor :#333333;
+$Lightgrey :#cccccc; 
+$Silver :#bfbfbf;
+
```
変数名の命名ルールを統一したいので、修正お願いします。

単語の間に-(ハイフン)をいれる 
例：Ash-gray

eclair6632 commented on app/assets/stylesheets/pages/_newpage_items_new.scss:
```diff
@@ -0,0 +1,245 @@
+$gray: #808080;
+$ashgrey: #888888;
+$safecolor :#333333;
+$Lightgrey :#cccccc; 
```
greyの綴りが違うと思います。grayかな？

hiroki-norota replied:
URLのLight greyから取ってきています。
https://www.color-sample.com/colors/cccccc/

eclair6632 commented on app/assets/stylesheets/pages/_newpage_items_new.scss:
```diff
@@ -0,0 +1,245 @@
+$WhiteSmoke: #f5f5f5;
+$cyan: #0095ee;
+$gray: #808080;
+$ashgrey: #888888;
```
ここも7行目と一緒で、綴りが間違っています。

hiroki-norota replied:
URL記載のAsh Greyから取ってきています。
https://www.color-sample.com/colors/888888/

eclair6632 replied:
aとeの違いは、アメリカ英語とイギリス英語の違いみたいです。
aで統一しませんか？

eclair6632 commented on app/assets/stylesheets/pages/_newpage_items_new.scss:
```diff
@@ -0,0 +1,245 @@
+  .select-box {
+    @include select-box;
+  }
+  .contents-hight {
```
このクラス名のhightは、heightの間違いでしょうか？

eclair6632 commented on app/assets/stylesheets/pages/_newpage_items_new.scss:
```diff
@@ -0,0 +1,245 @@
+        @include form-required;
+      }
+      &--details {
+        color: $ashgrey;
```
ここはあえてアッシュグレイにしているのでしょうか。
周りと合わせて黒に変更しませんか？

eclair6632 commented on app/views/items/_form_menu.html.haml:
```diff
@@ -0,0 +1,118 @@
+          
+      .delivery
+        %p.p-title
+          配送について
```
章節のくくりで考えると、33、34行目にある商品の詳細と同じレベルなので、
フォントのプロパティを合わせてみてはいかがでしょうか。

eclair6632 commented on app/views/items/_form_menu.html.haml:
```diff
@@ -0,0 +1,118 @@
+
+      .cost
+        %p.p-title
+          価格（¥300〜9,999,999）
```
章節のくくりで考えると、33、34行目にある商品の詳細と同じレベルなので、
フォントのプロパティを合わせてみてはいかがでしょうか。

hiroki-norota committed:
MU-new page fix 2

cost--selling__priceクラスに指定したbox-sizing: border-box;を削除。
form-newクラスの中にnew-page全体に適用するよう、設定しているため。

eclair6632 reviewed (commented):
コメントしました。ご確認ください。

shry44 commented on app/assets/stylesheets/pages/_newpage_items_new.scss:
```diff
@@ -0,0 +1,244 @@
+      }
+    }
+    
+    .items-date {
```
データの意味であればdata?

hiroki-norota replied:
deteだと日付でしたねw

hiroki-norota committed:
Merge branch 'MU_item-details' of https://github.com/jikutkgtaku/freemarket_sample_71d into MU_newpage

jikutkgtaku commented on app/assets/stylesheets/modules/_kategory-hierarchy.scss:
```diff
@@ -0,0 +1,25 @@
+// 上部のカテゴリ階層表示
+.kategory-hierarchy {
```
カテゴリーはcategoryですね。

jikutkgtaku commented:
誤ってmergeした影響なのか、ページにcssがあてがわれていません。少し修正しようとしたのですが、よくわからないので断念しました。

修正内容：application.scssに
@import "mixin/item-detail/_item-place";
@import "modules/_item-detail_items_show";

を追記。その後ページに戻るも、cssが反映されていませんでした。修正お願いします。

hiroki-norota committed:
MU-new page fix 3

newpageに@import設定
color name修正

hiroki-norota committed:
MU-new page fix 4

color nameをスネークケースに修正

shry44 commented:
LGTMです。categotyのスペルミスは私です。すみません。パンくず実装時にファイル名ごと変更になるため現時点では修正不要と思っています。

kanayan1225 commented:
私もLGTMです！

hiroki-norota committed:
MU-new page fix 5

page delete

hiroki-norota committed:
MU-new page fix 6

kategory-hierarchy file delete

hiroki-norota committed:
routing configuration

hiroki-norota committed:
Merge branch 'master' into MU_newpage

hiroki-norota commented:
商品出品ページのスクリーンショット
https://gyazo.com/d8295dc33f90032c4c3d1f6787f38652
https://gyazo.com/9a5f559bf6c0fa9db6a80d2a9a397f92
https://gyazo.com/f207aa59b709a7e18aaa04556a665952
https://gyazo.com/03912eb8197ea3499b5835c82cb13f93

i-date commented:
レビュー依頼＆スクショありがとうございます！
コンフリクトが生じているようですので、まずはコンフリクトを解消しましょう。
レビュー後にコンフリクトを修正すると、レビュー済みのコードが変化してしまいますので、次回からはレビュー依頼前にコンフリクトを解消していただければと思います。

hiroki-norota committed:
Merge branch 'master' into MU_newpage

miyagit reviewed (approved):
LGTM
